crates/paralegal-policy/src/context.rs

```diff
@@ -439,36 +438,43 @@ impl Context {
         let mut num_reached = 0;
         let mut num_checkpointed = 0;
 
-        let mut queue = starting_points.collect::<Vec<_>>();
+        // Package nodes with a reference to their controller's dataflow for efficiency.
+        struct NodeWithFlow<'a> {
```

Comment:
I was thinking of a simple tuple but of course this works too 😄

Reply:
Oops I forgot that those exist haha

Reply:
Too much Java? :P

Reply:
oh god no